### ISSUE TITLE
XcmV3Junctions uses XcmV4Junction instead of declaring a separate XcmV3Junction type

### ISSUE THREAD

**Issue by ryanleecode** [polkadot-api/polkadot-api#341]
The current codegen on fee5ecb6d09dea073f130fd2a58995907bd36c9c generates the following using westend asset hub as the metadata

```ts
export type XcmV3Junctions = Enum<{ type: "Here", value: undefined } | { type: "X1", value: Anonymize<XcmV4Junction> } | { type: "X2", value: Anonymize<I42l4nthiehb7> } | { type: "X3", value: Anonymize<I2jk9pdm4ajs0n> } | { type: "X4", value: Anonymize<I293rauivpnv4n> } | { type: "X5", value: Anonymize<Id42rc2s9m61aa> } | { type: "X6", value: Anonymize<Ibe9k3j6og3ch4> } | { type: "X7", value: Anonymize<I3vkvorkiqho0h> } | { type: "X8", value: Anonymize<Icmb7nn8ip4qrt> }>;
export const XcmV3Junctions = _Enum as unknown as GetEnum<XcmV3Junctions>;

export type XcmV4Junction = Enum<{ type: "Parachain", value: Anonymize<number> } | { type: "AccountId32", value: Anonymize<I5891blicehaji> } | { type: "AccountIndex64", value: Anonymize<Idrke3qhmim88u> } | { type: "AccountKey20", value: Anonymize<I3liki1s5lgett> } | { type: "PalletInstance", value: Anonymize<number> } | { type: "GeneralIndex", value: Anonymize<bigint> } | { type: "GeneralKey", value: Anonymize<Ic1rqnlu0a9i3k> } | { type: "OnlyChild", value: undefined } | { type: "Plurality", value: Anonymize<Ibb5u0oo9gtas> } | { type: "GlobalConsensus", value: Anonymize<XcmV4JunctionNetworkId> }>;
export const XcmV4Junction = _Enum as unknown as GetEnum<XcmV4Junction>;
```

`XcmV3Junctions` is constructed with `XcmV4Junction` rather than `XcmV3Junction`, which doesn't exist. Looking at the metadata, XcmV4Junction and XcmV3Junction are the same type.

https://www.diffchecker.com/ayqPJnXm/

Should we provide an alias for XcmV3Junction? Because I am writing client side code like this and it is strange seeing ` XcmV4Junction` inside of a  `XcmV3Junctions.X2` call.

```ts
 .submit$(account.address, {
        asset: {
          parents: 0,
          interior: XcmV3Junctions.X2([
            XcmV4Junction.PalletInstance(50),
            XcmV4Junction.GeneralIndex(BigInt(ASSET_ID)),
          ]),
        },
      })
```

Of course, one such solution is to put it on the user to just rename the import and then problem solved. question is, is this the desired DX? I have no idea so putting it up for discussion.

**Comment by josepot** [comment]
I fail to understand why this is an issue

**Comment by josepot** [comment]
> `XcmV3Junctions` is constructed with `XcmV4Junction` rather than `XcmV3Junction`, which doesn't exist.

It [does exist](https://github.com/polkadot-api/polkadot-api/blob/fee5ecb6d09dea073f130fd2a58995907bd36c9c/packages/codegen/src/known-types.ts#L700-L713)

> Looking at the metadata, XcmV4Junction and XcmV3Junction are the same type.

I don't think so.

**Comment by ryanleecode** [comment]
> > `XcmV3Junctions` is constructed with `XcmV4Junction` rather than `XcmV3Junction`, which doesn't exist.
> 
> It [does exist](https://github.com/polkadot-api/polkadot-api/blob/fee5ecb6d09dea073f130fd2a58995907bd36c9c/packages/codegen/src/known-types.ts#L700-L713)
> 
> > Looking at the metadata, XcmV4Junction and XcmV3Junction are the same type.
> 
> I don't think so.

run this https://github.com/ryanleecode/westend-asset-hub-test-papi

and u should get the following as the codegen

https://gist.github.com/ryanleecode/95c46612bf084c6c480234fb59a606ee

--

> I fail to understand why this is an issue

Doesn't have to be, depends how you view it ofc. For me I was updating your [example app ](https://github.com/josepot/westend-asset-hub-test-papi/blob/main/src/App.tsx#L12) and it was a bit of a headscratcher to see the `XcmV3Junction` vanished. I checked the types and saw that it was aliased with `XcmV4Junction`. And that left me wondering are `XcmV4Junction` and `XcmV3Junction` really the same, is the codegen bugged? So i checked the metadata and saw that it was. 

Now its obvious, well duh just `import XcmV4Junction as XcmV3Junction`. But it may not be obvious to **everyone**, especially newer developers or developers who are new to the ecosystem. For example a developer might do the following,

1. type `XcmV3Junction`
2. hit the quick fix hotkey in their editor, to make it auto import
3. no autoimport shows up
4. developer is now confused

![image](https://github.com/polkadot-api/polkadot-api/assets/21375952/b6868b94-7b12-4d79-b736-875d304b8b2c)


 So i'm not saying the current codegen is wrong, on a technical basis it is correct; just wondering if this is the right DX.

**Comment by josepot** [comment]
> So i checked the metadata and saw that it was.

It really isn't, though. I probably made a mistake when I did the initial seed of the dictionary... Although, the mistake is likely to be in the name of the `XcmV3Junctions`.

Ok, I see. yeah, this is an issue... but probably not what you thought it was. Ok, so what I meant is that in principle it's very possible that in a runtime where they have updated to a newer version of XCM, some of the inner primitives are still using constructs from previous versions... So, it's not necessarily wrong to see an Enum of a newer version use something from another version. However, what you are describing here it is off, because its an older version referencing something newer, and that doesn't make sense.

Again, this is likely because of a mistake that I made when i created the initial seed of the dictionary... Whenever we have some time we will improve both the dictionary, and the tooling that we use for maintaining it.

So, yeah, I would rather close this issue and open a more generic issue about that... Because that's something that's under our radar. Also, any help with that would be greatly appreciated.
